Nodeport loadbalanceing (#72)
UKCloud/openshift-v4: post-deployment/openstack/nodeport-loadbalancing-ingress/README.md

# NodePort loadbalancing using OpenStack Octavia

> This features enables customers to expose their NodePorts via a single public IP using OpenStack Octavia Loadbalancer.

Reviewer: ```suggestion
This feature enables customers to expose their NodePorts via a single public IP using OpenStack Octavia Loadbalancer.
```

**WARNING**: This is not supported for OpenStack Projects with multiple clusters.

## Prerequisite

Before you run this, you must first have a loadbalancer deployed in your OpenStack Environment. This can be done by running the following playbooks in order:

nodeport-loadbalance-network.yaml
nodeport-loadbalance-ingress.yaml

> The can be found in https://github.com/UKCloud/openshift-upi-ansible/tree/nodeport-loadbalancing

Reviewer: Should we be referencing a branch that is likely going to be deleted once merged?

# Note for v4.6

For OpenShift v4.6, it will be necessary to edit `cronjob-update-nodeport-loadbalancer.j2` to be `apiVersion: batch/v1beta1` before post-deploy is ran - this change is incompatible with v4.7+

# Method (for testing only)

1. Fill vars.yml with the information are you would normally.
2. Run the main.yaml playbook. Make sure you are logged into your cluster and have access to the OpenStack API for the project.
3. Once the deployment is completed, go to the "ukc-ingress" project and find the ConfigMap called "ukcloud-openstack-loadbalancer".
4. Create your NodePort service that you want your application to use. 
4. Add your loadbalancer settings to the data section of the ConfigMap. Here is an example:

```yaml
data:
  openstack: |
    - nodeport: 30418
      listener: 1234
      allowed_ips:
        - 0.0.0.0/0
        - 80.81.82.83/24
```

> nodeport = the port that the NodePort service is exposing on. 

Reviewer: ```suggestion
nodeport = the port that the NodePort service is exposed on.
```

listener = the port that you want the loadbalancer to listen on.

> allowed_ips = list of source CIDR's that you want the loadbalanacer to only accept traffic from.

Reviewer: ```suggestion
allowed_ips = list of source CIDRs that you want the loadbalancer to only accept traffic from.
```
